src/components/BaseComponents/BaseMarker.tsx

```diff
@@ -9,7 +9,7 @@ const DivIconMarker = dynamic(
 
 export type BaseMarkerProps = {
     position: [number, number];
-    handleClick: any;
+    handleClick?: any;
```

Comment:
do we need this at all?

Reply:
Yeah, it detects the click event on the marker. If you have any alternative, sure we can get rid off it.

Reply:
sorry I didn't explain my self at all here. I was thinking that this method seems to be very specific and that maybe it doesn't need to be a prop, but a util method. So we don't have to pass it there as an arg to the BaseMarker, but call it straight when needed.

Reply:
Let's discuss this offline as well.

Reply:
Hmmm, I had another look at it after yesterday's conversation. Converting this to a util and not passing a prop for handleClick will just make things alot more complicated.
`const handleModal = (value: boolean, stationId: number) => {
        stationsProvider.handleModal(value);
        setActiveStation(stationId);
    };`

HandleModal is setting local state for the HomepageMap component. This state will have to be passed down to the Marker, however it is not relevant to the marker at all. Also I am not very fond of bringing the provider state to the Marker as well. I think the best approach to this, is to be refactored and become a real Base component and extend the Modal functionality into a different one (i.e. MarkerWithModal). I am going to add a reminder for this, to be handled in the future. What do you think?

Reply:
Yep that makes sense, would you like to create the refactoring ticket and add a TO-DO comment about it so we can leave it as is for now?

Reply:
Yeah, I have added the todo comment and I am going to create the ticket and add it in the comment as well.